Reprotest for deb packages is failing

Opening issue based on report in https://github.com/freedomofpress/securedrop-debian-packaging/pull/275 In short, the `reprotest-debs` test is failing on that branch. Is it failing elsewhere? Hard to say: we don't run that check as part of the nightly tests, but we sure could do so. 

Here's an example failing test: https://app.circleci.com/pipelines/github/freedomofpress/securedrop-debian-packaging/1343/workflows/1a0b16f4-d241-4057-8f34-33c99a1ce8e7/jobs/10448 and failure output:

```
PKG_VERSION not set, inferring from recent releases...
Failed to infer version
make: *** [Makefile:5: securedrop-proxy] Error 1
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/reprotest/__init__.py", line 831, in run
    return 0 if check_func(*check_args) else 1
  File "/usr/lib/python3/dist-packages/reprotest/__init__.py", line 363, in check
    local_dists = [proc.send(nv) for nv in zip(bnames, build_variations)]
  File "/usr/lib/python3/dist-packages/reprotest/__init__.py", line 363, in <listcomp>
    local_dists = [proc.send(nv) for nv in zip(bnames, build_variations)]
  File "/usr/lib/python3/dist-packages/reprotest/__init__.py", line 329, in corun_builds
    bctx.run_build(testbed, build, os.environ, artifact_pattern, testbed_build_pre, no_clean_on_error)
  File "/usr/lib/python3/dist-packages/reprotest/__init__.py", line 220, in run_build
    kind='build')
  File "/usr/lib/python3/dist-packages/reprotest/__init__.py", line 64, in check_exec2
    adtlog.AutopkgtestError)
  File "/usr/lib/python3/dist-packages/reprotest/__init__.py", line 70, in bomb
    raise _type(m)
reprotest.lib.adtlog.AutopkgtestError: "sh -ec run_build() {
    mkdir -p /tmp/reprotest.p0VvwL/build-control-aux && \
    mv /tmp/reprotest.p0VvwL/build-control/ /tmp/reprotest.p0VvwL/const_build_path && \
    SETARCH_ARCH=$(uname -m) && \
    SETARCH_OPTS="$SETARCH_OPTS -R" && \
    CPU_MAX=$(nproc) && \
    CPU_MIN=$({ echo $CPU_MAX; echo 1; } | sort -n | head -n1) && \
    CPU_NUM=$CPU_MIN && \
    umask 0022 && \
    export REPROTEST_BUILD_PATH=/tmp/reprotest.p0VvwL/const_build_path/ && \
    export REPROTEST_UMASK=$(umask) && \
    faketime @1635792985 \
    taskset -a -c $(echo $(shuf -i0-$((CPU_MAX - 1)) -n$CPU_NUM) | tr ' ' ,) \
     \
    sh -ec 'cd "$REPROTEST_BUILD_PATH"; unset REPROTEST_BUILD_PATH; umask "$REPROTEST_UMASK"; unset REPROTEST_UMASK; make securedrop-proxy'
}
```

--- a/scripts/build-debianpackage
+++ b/scripts/build-debianpackage
@@ -39,7 +39,7 @@ fi
 
 # Look up most recent release from GitHub repo
 function find_latest_version() {
-    repo_url="https://github.com/freedomofpress/${PKG_NAME}/releases"
+    repo_url="https://github.com/freedomofpress/${PKG_NAME}/tags"
     curl -s "$repo_url" \
         | perl -nE '$_ =~ m#/releases/tag/(v?[\d\.]+)\"# and say $1' \
         | head -n 1
@@ -67,9 +67,9 @@ function verify_git_tag() {
     d="$1"
     t="$2"
     prod_fingerprint="2359E6538C0613E652955E6C188EDD3B7B22E6A3"
-    if ! git -C "$build_dir" tag --verify "$PKG_VERSION" 2>&1 \
+    if ! git -C "$d" tag --verify "$t" 2>&1 \
         | grep -q -F "using RSA key $prod_fingerprint" ; then
-        echo "Failed to verify $PKG_VERSION, not signed with $prod_fingerprint" >&2
+        echo "Failed to verify $t, not signed with $prod_fingerprint" >&2
         exit 2
     fi
 }
@@ -98,8 +98,8 @@ function build_source_tarball() {
     # Initial tarball will contain timestamps from NOW, let's repack
     # with timestamps from the changelog, which is static.
     raw_tarball="$(find "${build_dir}/dist/" | grep -P '\.tar.gz$' | head -n1)"
-    dch_time="$(date "+%Y-%m-%d %H:%M:%S %z" -d@$(dpkg-parsechangelog --file $PKG_NAME/debian/changelog-$PLATFORM -STimestamp)) "
-    (cd "$build_dir" && tar -xzf "dist/$(basename $raw_tarball)")
+    dch_time="$(date "+%Y-%m-%d %H:%M:%S %z" -d@"$(dpkg-parsechangelog --file "${PKG_NAME}/debian/changelog-${PLATFORM}" -STimestamp)") "
+    (cd "$build_dir" && tar -xzf "dist/$(basename "$raw_tarball")")
     tarball_basename="$(basename "$raw_tarball")"
     # Repack with tar only, so env vars are respected
     (cd "$build_dir" && tar -cf "${tarball_basename%.gz}" --mode=go=rX,u+rw,a-s --mtime="$dch_time" --sort=name --owner=root:0 --group=root:0 "${tarball_basename%.tar.gz}" 1>&2)
@@ -118,7 +118,7 @@ if [[ "${PKG_NAME}" =~ ^(securedrop-client|securedrop-proxy|securedrop-export|se
         # Build from source
         echo "PKG_PATH not set, building from source (version $PKG_VERSION)..."
         build_source_tarball
-        candidate_pkg_path="$(find /tmp/$PKG_NAME/dist -type f -iname '*.tar.gz')"
+        candidate_pkg_path="$(find "/tmp/${PKG_NAME}/dist" -type f -iname '*.tar.gz')"
         if [[ -f "$candidate_pkg_path" ]]; then
             PKG_PATH="$candidate_pkg_path"
             echo "Found tarball at $PKG_PATH, override with PKG_PATH..."
@@ -138,7 +138,7 @@ if [[ "${PKG_NAME}" =~ ^(securedrop-client|securedrop-proxy|securedrop-export|se
     cd "$TOP_BUILDDIR/$PKG_NAME/"
 
     # Verify all the hashes from the verified sha256sums.txt
-    $CUR_DIR/scripts/verify-hashes $CUR_DIR/sha256sums.txt
+    "${CUR_DIR}/scripts/verify-hashes" "${CUR_DIR}/sha256sums.txt"
 
     echo "All hashes verified."
 else
@@ -153,7 +153,8 @@ echo "$TOP_BUILDDIR/$PKG_NAME/"
 mv "$TOP_BUILDDIR/$PKG_NAME/debian/changelog-$PLATFORM" "$TOP_BUILDDIR/$PKG_NAME/debian/changelog"
 
 # Adds reproducibility step
-export SOURCE_DATE_EPOCH=`dpkg-parsechangelog -STimestamp`
+SOURCE_DATE_EPOCH="$(dpkg-parsechangelog -STimestamp)"
+export SOURCE_DATE_EPOCH
 
 # Build the package
 dpkg-buildpackage -us -uc

